.github/workflows/release_drafter.yaml

```diff
@@ -56,4 +56,33 @@ jobs:
           publish: "${{ steps.tag-version.outputs.new_tag != '' }}"
           tag: "${{ steps.tag-version.outputs.new_tag }}"
         env:
-          GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
\ No newline at end of file
+          GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
+
+  deploy:
+    # This will upload a Python Package using Twine when a release is created
+    # For more information see: https://help.github.com/en/actions/language-and-framework-guides/using-python-with-github-actions#publishing-to-package-registries
+    #
+    # This job will run when you have tagged a commit, starting with "v*"
+    # or created a release in GitHub which includes a tag starting with "v*"
+    # and requires that you have put your twine API key in your
+    # github secrets (see readme for details)
+    needs: [update_release_draft]
+    runs-on: ubuntu-latest
+    if: contains(github.ref, 'tags')
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.x'
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install build twine
+    - name: Build and publish
+      env:
+        TWINE_USERNAME: ${{ secrets.PYPI_USERNAME }}
```

Comment:
@constantinpape will need to set up the `PYPI_USERNAME` and `PYPI_PASSWORD` github secrets before we can merge this PR. These should be new API tokens for PyPI, not your personal username and password.